calico/felix/frules.py

```diff
@@ -574,12 +574,18 @@ def update_ipsets(type,
     """
     for rule in rule_list:
         if rule.get('cidr') is None:
-            log.error("Invalid %s rule without cidr for %s : %s" %
-                      (descr, suffix, rule))
+            log.error("Invalid %s rule without cidr for %s : %s",
+                      descr, suffix, rule)
+            continue
+
+        if (type == IPV4 and not common.validate_cidr(rule['cidr'], 4) or
+            type == IPV6 and not common.validate_cidr(rule['cidr'], 6)    ):
```

Comment:
Might be wise to bracket each of these conditional sections. Suspect most people don't know operator precedence order well enough to be confident of what will happen here.